sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java

```diff
@@ -94,14 +131,18 @@ private Duration determineDelayDuration(HttpResponse response, int tryCount) {
             return retryStrategy.calculateRetryDelay(tryCount);
         }
 
-        String retryHeader = response.getHeaderValue(RETRY_AFTER_MS_HEADER);
+        String retryHeaderValue = null;
+
+        if (!isNullOrEmpty(retryAfterHeader)) {
+            retryHeaderValue = response.getHeaderValue(retryAfterHeader);
```

Comment:
retryAfterHeader : This could be null  when user do not want to use any response header for retry.